views/js/controller/runner/runner.js

```diff
@@ -174,6 +174,11 @@ define([
 
             // adding attr for RTL languages
             $('.delivery-scope').attr({dir: locale.getLanguageDirection(context.locale)});
+            // tune classes to remove page header RTL artifacts
+            const menuItemSelector = '.delivery-scope[dir=rtl] .settings-menu li';
+            $(menuItemSelector).addClass('sep-before');
+            $(`${menuItemSelector}:visible:last`).removeClass('sep-before');
```

Comment:
**nitpick:** The name is a bit confusing. It should rather highlight the relationship with RTL

**suggestion:** Instead of calling it `menuItemSelector`, better to name it `menuItemSelectorRTL` or something like that.

Reply:
this is smth I want to try as well to use only CSS, if possible.

Reply:
However, the CSS won't remove the visual glitch, as the RTL direction is set by the FE.
A better approach, would be definitely to set it from the BE directly on the template.

This may need some help from BE guys though. Eventually I can help adding the missing part.